lib/view_component/collection.rb

```diff
@@ -26,5 +28,24 @@ def collection_variable(object)
         raise ArgumentError.new("The value of the argument isn't a valid collection. Make sure it responds to to_ary: #{object.inspect}")
       end
     end
+
+    def collection_parameter_name
+      @component.collection_parameter_name
+    end
+
+    def collection_counter_parameter_name
+      "#{collection_parameter_name}_counter".to_sym
+    end
+
+    def component_options(item, iterator)
+      item_options = { collection_parameter_name => item }
+      item_options[collection_counter_parameter_name] = iterator.index + 1 if counter_argument_present?
+
+      @options.merge(item_options)
+    end
+
+    def counter_argument_present?
+      @counter_argument_present ||= @component.instance_method(:initialize).parameters.map(&:second).include?(collection_counter_parameter_name)
```

Comment:
Why does it matter if the argument is `second`? What happens if we have many arguments to the component? 

Generally, Ruby kwargs should be not be order-dependent. I'd like to avoid introducing such a dependency here if possible.

Reply:
Hi @joelhawksley 

This is not with the order fo the arguments, it's because `instance_method(:initialize).parameters` returns a nested array and the second array item is the one that holds the argument name. 

Here is an example using my own code
```
$ Atoms::AvatarComponent.instance_method(:initialize).parameters

[[:keyreq, :user], [:key, :size], [:key, :glyph], [:key, :color], [:key, :html]]
```

Reply:
🤦 I I misread the code! Thank you for clarifying ❤️